docs: Improve doc about TLS (#1019)
redpanda-data/documentation: versioned_docs/version-22.3/manage/kubernetes/configure-storage.mdx

---
title: Configure Storage in Kubernetes
tags:
  - Kubernetes
  - Helm configuration
---

<head>
    <meta name="title" content="Configure Storage in Kubernetes | Redpanda Docs"/>
    <meta name="description" content="Configure the Helm chart to use PersistentVolumes, hostPath volumes, or emptyDir volumes."/>
</head>

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import ValuesYaml from './shared/_values-yaml.mdx'

Redpanda brokers must store their data on disk. By default, the Redpanda Helm chart uses the default StorageClass in a Kubernetes cluster to create one PersistentVolumeClaim for each Pod that runs a Redpanda broker. The default StorageClass in your Kubernetes cluster depends on the Kubernetes platform that you are using. You can customize the Helm chart to use the following storage types:

- [PersistentVolumes](#use-persistentvolumes)
- [hostPath volumes](#use-hostpath-volumes)
- [emptyDir volumes](#use-emptydir-volumes)

## Prerequisites

> - If you're configuring Redpanda for production, make sure to create and mount an **XFS file system** on any storage volumes that host the data directory of Redpanda (`/var/lib/redpanda/data`). XFS is a high-performance file system that is required for running Redpanda in production. NFS file systems are not supported.

Reviewer: ```suggestion
- If you're configuring Redpanda for production, you must create and mount an **XFS file system** on any storage volumes that host the data directory of Redpanda (`/var/lib/redpanda/data`). XFS is a high-performance file system that is required for running Redpanda in production. NFS file systems are not supported.
```

You can configure the Helm chart to use PersistentVolumes with a static provisioner or a dynamic provisioner.
Redpanda recommends using a StorageClass with a dynamic provisioner. See the [best practices](../../../deploy/deployment-option/self-hosted/kubernetes/kubernetes-best-practices/).

<Tabs groupId="provisioners">
<TabItem value="dynamic" label="Dynamic provisioners">

A dynamic provisioner creates a PersistentVolume on demand for each Redpanda broker.

Managed Kubernetes platforms and cloud environments usually provide a dynamic provisioner.

> If you are running Kubernetes on-premises, you may need to deploy a dynamic provisioner for your storage type.

Reviewer: I read this and think, well, when do I have to and when do I not have to? "you may" immediately triggers "you may not" in my head, and I wonder what the trigger is for each case.
